### ISSUE TITLE
Drawing the cam to a PGraphics breaks the library

### ISSUE THREAD

**Issue by clankill3r** [processing/processing-video#112]
This is the getting started capture example, slightly modified.
I added a PGraphics and draw the cam to there.
After that cam.available() will never equal to true anymore.

I added a line in the code:
`//image(cam, 0, 0, width, height); // << uncommenting this will fix the issue`
A draw to the main graphics before drawing to another PGraphics is fixing the issue.

This happens on OSX.
For me on 10.12.6 with processing 3.5.3.

I also tested this on another mac, with more likely a newer OS and there was also the problem.

```
/**
 * Getting Started with Capture.
 * 
 * Reading and displaying an image from an attached Capture device. 
 */

import processing.video.*;

Capture cam;


PGraphics pg;

void setup() {
  size(640, 480);

  String[] cameras = Capture.list();

  if (cameras == null) {
    println("Failed to retrieve the list of available cameras, will try the default...");
    cam = new Capture(this, 640, 480);
  } if (cameras.length == 0) {
    println("There are no cameras available for capture.");
    exit();
  } else {
    println("Available cameras:");
    printArray(cameras);

    // The camera can be initialized directly using an element
    // from the array returned by list():
    cam = new Capture(this, cameras[0]);
    // Or, the settings can be defined based on the text in the list
    //cam = new Capture(this, 640, 480, "Built-in iSight", 30);
    
    // Start capturing the images from the camera
    cam.start();
  }
  
  pg = createGraphics(width, height);
  pg.beginDraw();
  pg.endDraw();
}




void draw() {
  if (cam.available() == true) {
    cam.read();
    pg.beginDraw();
    //image(cam, 0, 0, width, height); // << uncommenting this will fix the issue
    pg.image(cam, 0, 0);
    pg.endDraw();
  }
  image(cam, 0, 0, width, height);
  // The following does the same as the above image() line, but 
  // is faster when just drawing the image without any additional 
  // resizing, transformations, or tint.
  //set(0, 0, cam);
}
```

**Comment by codeanticode** [comment]
@clankill3r Is this happening with the latest beta of the v2 of the library?

**Comment by clankill3r** [comment]
Yeah just tested, same problem.

Also the available cameras becomes:

```
Available cameras:
[0] "0"
```
With an red error:

> Device enumeration is currently not supported on your platform.

**Comment by codeanticode** [comment]
@clankill3r the latest beta release, [beta3](https://github.com/processing/processing-video/releases/tag/r5-v2.0-beta3), fixed device enumeration on mac, did you try that out?

**Comment by clankill3r** [comment]
@codeanticode 

I have this now:

```
Available cameras:
[0] "FaceTime HD Camera (Built-in)"
```

Not the whole list as before, but that was probably with software downscaling anyway?

**Comment by codeanticode** [comment]
Yes, that's correct, the new version of the library no longer lists all the native resolutions supposedly supported by the device. You can set any resolution you want, GStreamer will automatically resolve whether it requires any down/upscaling.

**Comment by codeanticode** [comment]
@clankill3r does software downscaling work for you? I tried capturing at low-res from an HD webcam, and seemed fine.

**Comment by clankill3r** [comment]
It works for me, it is slow with the default renderer and fast in opengl.

Op zo 25 aug. 2019 om 16:25 schreef codeanticode <notifications@github.com>:

**Comment by clankill3r** [comment]
The first issue still exist.

Here is another sketch to show it.

`import processing.video.*;

Capture cam;

int BUFFER_SIZE = 60 * 3;

PGraphics[] buffer = new PGraphics[BUFFER_SIZE];
int index = 0;


void setup() {
  size(640, 480);

  String[] cameras = Capture.list();

  if (cameras == null) {
    println("Failed to retrieve the list of available cameras, will try the default...");
    cam = new Capture(this, 640, 480);
  } if (cameras.length == 0) {
    println("There are no cameras available for capture.");
    exit();
  } else {
    println("Available cameras:");
    printArray(cameras);

    // The camera can be initialized directly using an element
    // from the array returned by list():
    cam = new Capture(this, cameras[0]);
    // Or, the settings can be defined based on the text in the list
    //cam = new Capture(this, 640, 480, "Built-in iSight", 30);
    
    // Start capturing the images from the camera
    cam.start();
  }
  
  for (int i = 0; i < BUFFER_SIZE; i++) {
    PGraphics g = createGraphics(640, 480);
    g.beginDraw();
    g.background(0);
    g.endDraw();
    buffer[i] = g;
  }
  
  
}

void draw() {
  if (cam.available() == true) {
    cam.read();
    
    PGraphics g = buffer[index];
    
    g.beginDraw();
    g.image(cam, 0, 0);
    g.endDraw();
    
    index++;
    if (index == BUFFER_SIZE-1) {
      index = 0;  
    }
    
  }
  image(cam, 0, 0, width, height);
  // The following does the same as the above image() line, but 
  // is faster when just drawing the image without any additional 
  // resizing, transformations, or tint.
  //set(0, 0, cam);
  
  
  int delay_n_frames = (int) map(mouseX, 0, width, 0, BUFFER_SIZE-1);
  
  fill(255);
  text(""+delay_n_frames, 250, 50);
  
  
  
  int show_index = index - delay_n_frames;
  if (show_index < 0) {
    show_index += BUFFER_SIZE;  
  }
  
  image(buffer[show_index], 0, 0, 100, 100);
  
  
}`
